src/lib/emptyingSet.js

```diff
@@ -0,0 +1,24 @@
+ /**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+module.exports = function emptyingSet() {
```

Comment:
I don't think this name is good. It should also be capitalized.

`SetWithEmptyPromise` or something might be more appropriate. It should also be created using `new`. Actually, can we just subclass Set instead of the hackery? :)